torch/csrc/MemoryFormat.cpp

```diff
@@ -0,0 +1,80 @@
+#include <torch/csrc/MemoryFormat.h>
+
+#include <torch/csrc/Exceptions.h>
+#include <torch/csrc/utils/object_ptr.h>
+#include <torch/csrc/utils/python_strings.h>
+
+#include <c10/core/MemoryFormat.h>
+
+#include <structmember.h>
+#include <cstring>
+#include <string>
+
+PyObject *THPMemoryFormat_New(at::MemoryFormat memory_format, const std::string& name)
+{
+  auto type = (PyTypeObject*)&THPMemoryFormatType;
+  auto self = THPObjectPtr{type->tp_alloc(type, 0)};
+  if (!self) throw python_error();
+  auto self_ = reinterpret_cast<THPMemoryFormat*>(self.get());
+  self_->memory_format = memory_format;
+  std::strncpy (self_->name, name.c_str(), MEMORY_FORMAT_NAME_LEN);
+  self_->name[MEMORY_FORMAT_NAME_LEN] = '\0';
+  return self.release();
+}
+
+PyObject *THPMemoryFormat_repr(THPMemoryFormat *self)
+{
+  return THPUtils_packString(self->name);
+}
+
+PyTypeObject THPMemoryFormatType = {
```

Comment:
btw, have we considered switching to pybind11 for enums like this? that's a lot of boilerplate